packages/flutter_tools/lib/src/build_runner/web_compilation_delegate.dart

```diff
@@ -7,205 +7,72 @@ import 'dart:async';
-          'profile': buildMode == BuildMode.profile,
-        },
-        'flutter_tools:test_entrypoint': <String, dynamic>{
-          'release': buildMode == BuildMode.release,
-          'profile': buildMode == BuildMode.profile,
-        },
-      },
-      isReleaseBuild: false,
-    );
-    return _builder.run(
-      const <AssetId, ChangeType>{},
-      buildDirs: buildDirs,
-    );
-  }
-
-  Future<void> _cleanAssets(Directory projectDirectory) async {
-    final File assetGraphFile = fs.file(core.assetGraphPath);
-    AssetGraph assetGraph;
-    try {
-      assetGraph = AssetGraph.deserialize(await assetGraphFile.readAsBytes());
-    } catch (_) {
-      printTrace('Failed to clean up asset graph.');
-    }
-    final core.PackageGraph packageGraph = core.PackageGraph.forThisPackage();
-    await _cleanUpSourceOutputs(assetGraph, packageGraph);
-    final Directory cacheDirectory = fs.directory(fs.path.join(
+      .createSync();
+    final BuildDaemonClient client = await buildDaemonCreator.startBuildDaemon(
       projectDirectory.path,
-      '.dart_tool',
-      'build',
-      'flutter_web',
-    ));
-    if (assetGraphFile.existsSync()) {
-      assetGraphFile.deleteSync();
-    }
-    if (cacheDirectory.existsSync()) {
-      cacheDirectory.deleteSync(recursive: true);
-    }
-  }
-
-  Future<void> _cleanUpSourceOutputs(AssetGraph assetGraph, core.PackageGraph packageGraph) async {
-    final core.FileBasedAssetWriter writer = core.FileBasedAssetWriter(packageGraph);
-    if (assetGraph?.outputs == null) {
-      return;
-    }
-    for (AssetId id in assetGraph.outputs) {
-      if (id.package != packageGraph.root.name) {
-        continue;
+      release: mode == BuildMode.release,
+      profile: mode == BuildMode.profile,
+      hasPlugins: false,
+      includeTests: true,
+    );
+    client.startBuild();
+    bool success = true;
+    await for (BuildResults results in client.buildResults) {
+      final BuildResult result = results.results.firstWhere((BuildResult result) {
+        return result.target == 'web';
+      });
+      if (result.status == BuildStatus.failed) {
+        success = false;
+      }
+      if (result.status == BuildStatus.succeeded) {
+        break;
       }
-      final GeneratedAssetNode node = assetGraph.get(id);
-      if (node.wasOutput) {
-        // Note that this does a file.exists check in the root package and
-        // only tries to delete the file if it exists. This way we only
-        // actually delete to_source outputs, without reading in the build
-        // actions.
-        await writer.delete(id);
+    }
+    if (success && testOutputDir != null) {
+      final Directory rootDirectory = projectDirectory
+        .childDirectory('.dart_tool')
+        .childDirectory('build')
+        .childDirectory('flutter_web');
+
+      for (Directory childDirectory in rootDirectory
+        .listSync()
+        .whereType<Directory>()) {
```

Comment:
Maybe extract the iterable into a variable?

Reply:
Reformatted